lib/game_teams_manager.rb

```diff
@@ -112,33 +109,24 @@ def rival(team_id)
     rival_hash_ratio.key(rival_hash_ratio.values.max)
   end
 
-  def total_goals_by_team
-    goals_by_team_id = Hash.new(0)
-    game_teams.each do |game|
-      goals_by_team_id[game.team_id] += game.goals
-    end
-    goals_by_team_id
-  end
-
-  def total_games_by_team
-    games_by_team_id = Hash.new(0)
-    game_teams.each do |game|
-      games_by_team_id[game.team_id] += 1
+  def create_goals_hash
+    goals_hash = Hash.new { |coach, team| coach[team] = [0,0] }
+    @game_teams.each do |game_team|
+      goals_hash[game_team.team_id][1] += 1
+      goals_hash[game_team.team_id][0] += game_team.goals
```

Comment:
If I am understanding correctly create_goals_hash replaces total_goals and total_games_by_team?

Reply:
yep. that's right